src/parser/jobs/ast/modules/ArcanaTracking/ArcanaTracking.tsx

```diff
@@ -181,45 +176,43 @@ export default class ArcanaTracking extends Module {
 		}
 
 		this.prepullArcanas.forEach(arcanaBuff => {
-			if (arcanaBuff.ability.guid === event.ability.guid
-				&& arcanaBuff.targetID === event.targetID) {
-
-				const cardStateItem: CardState = {..._.last(this.cardStateLog)} as CardState
-				const arcanaAction = getDataBy(ACTIONS, 'id', this.arcanaStatusToPlay(event.ability.guid))
-
-				if (arcanaAction) {
-					const arcanaCastEvent: CastEvent = {
-						ability: {
-							name: arcanaAction.name,
-							guid: arcanaAction.id as number,
-							type: 1,
-							abilityIcon: _.replace(_.replace(arcanaAction.icon, 'https://xivapi.com/i/', ''), '/', '-'),
-						},
-						timestamp: event.timestamp - (STATUSES.THE_BALANCE.duration * 1000),
-						type: 'cast',
-						sourceIsFriendly: true,
-						targetIsFriendly: true,
-						sourceID: event.sourceID,
-						targetID: event.targetID,
-					}
-					cardStateItem.lastEvent = {...arcanaCastEvent}
-					cardStateItem.drawState = DrawnType.NOTHING
-					cardStateItem.sealState = CLEAN_SEAL_STATE
-
-					this.cardStateLog.unshift(cardStateItem)
-					this.pullIndex++
-				}
-
-				return
+			if (!(arcanaBuff.ability.guid === event.ability.guid
+				&& arcanaBuff.targetID === event.targetID)) { return }
+
+			const cardStateItem: CardState = {..._.last(this.cardStateLog)} as CardState
+			const arcanaAction = getDataBy(ACTIONS, 'id', this.arcanaStatusToPlay(event.ability.guid))
+
+			if (!arcanaAction) { return }
+
+			const arcanaCastEvent: CastEvent = {
+				ability: {
+					name: arcanaAction.name,
+					guid: arcanaAction.id as number,
+					type: 1,
+					abilityIcon: _.replace(_.replace(arcanaAction.icon, 'https://xivapi.com/i/', ''), '/', '-'),
+				},
+				timestamp: event.timestamp - (STATUSES.THE_BALANCE.duration * 1000),
+				type: 'cast',
+				sourceIsFriendly: true,
+				targetIsFriendly: true,
+				sourceID: event.sourceID,
+				targetID: event.targetID,
 			}
+			cardStateItem.lastEvent = {...arcanaCastEvent}
```

Comment:
nit; spreading is mainly best for when you want a copy of the object/array so you don't/can't modify the original - given the original in this case is being constructed right above, that's probably not a concern.